corehq/apps/data_interfaces/views.py

```diff
@@ -509,11 +522,44 @@ def post(self, request, *args, **kwargs):
             )
         )
 
+    def inaccessible_forms_accessed(self, xform_ids, domain, couch_user):
+        xforms = FormAccessors(domain).get_forms(xform_ids)
+        xforms_user_ids = set([xform.user_id for xform in xforms])
+        accessible_user_ids = set(user_ids_at_accessible_locations(domain, couch_user))
+        return xforms_user_ids - accessible_user_ids
+
+    def get_xform_ids(self, request):
+        form_ids_or_query_string = self.get_form_ids_or_query_string(request)
```

Comment:
I just looked up the definition for this:
```python
    def get_form_ids_or_query_string(self, request):
        if 'select_all' in self.request.POST:
            # Altough evaluating form_ids and sending to task would be cleaner,
            # heavier calls should be in in an async task instead
            import urllib
            form_query_string = urllib.unquote(self.request.POST.get('select_all'))
            return form_query_string
        else:
            return self.request.POST.getlist('xform_ids')
```
It forks depending on parameters, then returns one value, forcing you to check the type of the return again here.  It might be more straightforward to just merge `get_form_ids_or_query_string` in to this method and only do one if/else block (so you can drop the `if type()==list` and `isinstance(basestring)`)